lab-express-basic-auth/routes/site-routes.js

```diff
@@ -0,0 +1,27 @@
+const express = require("express"); 
+
+const router = express.Router();
+
+router.get("/", (req,res,next) => {
+    res.render("home");
+});
+
+
+//Initial Route
+router.use ((req, res, next) => {
+    if (req.session.currentUser) {
+        next(); 
+    } else {
+        res.redirect ("/login")
+    }
+})
+
+
+
+//Next Route
+router.get("/member", (req,res,next) => {
+    res.render("member");
+});
+
```

Comment:
For the third iteration, you can write a middleware function and then pass it to the routes that need to be protected like for instance
```
const auth = (req, res, next) => {
  if (!req.session.user) {
    res.redirect("/login");
    return;
  }
  next();
};

router.get("/private", auth, (req, res, next) => {
  res.render("private");
});

router.get("/main", auth, (req, res, next) => {
  res.render("main");
});